exchanges/deribit/deribit_websocket_eps.go

```diff
@@ -0,0 +1,2174 @@
+	}
+	input := &struct {
+		Instrument string `json:"instrument_name"`
+	}{
+		Instrument: instrument,
+	}
+	var resp ContractSizeData
+	return &resp, d.SendWSRequest(getContractSize, input, &resp, false)
+}
+
+// WSRetriveCurrencies retrives all cryptocurrencies supported by the API through the websocket connection.
+func (d *Deribit) WSRetriveCurrencies() ([]CurrencyData, error) {
+	var resp []CurrencyData
+	return resp, d.SendWSRequest(getCurrencies, nil, &resp, false)
+}
+
+// WSRetriveDeliveryPrices retrives delivery prices using index name through the websocket connection.
+func (d *Deribit) WSRetriveDeliveryPrices(indexName string, offset, count int64) (*IndexDeliveryPrice, error) {
+	indexNames := map[string]bool{"ada_usd": true, "avax_usd": true, "btc_usd": true, "eth_usd": true, "dot_usd": true, "luna_usd": true, "matic_usd": true, "sol_usd": true, "usdc_usd": true, "xrp_usd": true, "ada_usdc": true, "avax_usdc": true, "btc_usdc": true, "eth_usdc": true, "dot_usdc": true, "luna_usdc": true, "matic_usdc": true, "sol_usdc": true, "xrp_usdc": true, "btcdvol_usdc": true, "ethdvol_usdc": true}
+	if !indexNames[indexName] {
+		return nil, errUnsupportedIndexName
+	}
+	input := &struct {
+		IndexName string `json:"index_name"`
+		Offset    int64  `json:"offset,omitempty"`
+		Count     int64  `json:"count,omitempty"`
+	}{
+		IndexName: indexName,
+		Offset:    offset,
+		Count:     count,
+	}
+	var resp IndexDeliveryPrice
+	return &resp, d.SendWSRequest(getDeliveryPrices, input, &resp, false)
+}
+
+// WSRetriveFundingChartData retrives funding chart data for the requested instrument and time length through the websocket connection.
+// supported lengths: 8h, 24h, 1m <-(1month)
+func (d *Deribit) WSRetriveFundingChartData(instrument, length string) (*FundingChartData, error) {
+	if instrument == "" {
+		return nil, fmt.Errorf("%w, instrument_name is missing", errInvalidInstrumentName)
+	}
+	if length != "8h" && length != "24h" && length != "12h" && length != "1m" {
+		return nil, fmt.Errorf("%w, only 8h, 12h, 1m, and 24h are supported", errIntervalNotSupported)
+	}
+	input := &struct {
+		InstrumentName string `json:"instrument_name"`
+		Length         string `json:"length"`
+	}{
+		InstrumentName: instrument,
+		Length:         length,
+	}
+	var resp FundingChartData
+	return &resp, d.SendWSRequest(getFundingChartData, input, &resp, false)
+}
+
+// WSRetriveFundingRateValue retrives funding rate value data through the websocket connection.
+func (d *Deribit) WSRetriveFundingRateValue(instrument string, startTime, endTime time.Time) (float64, error) {
+	if instrument == "" {
+		return 0, fmt.Errorf("%w, instrument_name is missing", errInvalidInstrumentName)
+	}
+	if startTime.IsZero() {
+		return 0, fmt.Errorf("%w, start time can not be zero", errInvalidTimestamp)
+	} else if startTime.After(endTime) {
+		return 0, errStartTimeCannotBeAfterEndTime
+	}
+	if endTime.IsZero() {
+		endTime = time.Now()
+	}
```

Comment:
This is an invalid check. Please use a common validation function that is correct